File: rides-service/src/main/java/com/example/rides_service/client/PassengerServiceClientFallback.java
package com.example.rides_service.client;

import com.example.rides_service.util.ExceptionMessages;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Component;

@Component
@Slf4j
public class PassengerServiceClientFallback implements PassengerServiceClient {
    @Override
    public ResponseEntity<Boolean> doesPassengerExists(Long passengerId) {
        throw new RuntimeException(ExceptionMessages.UNABLE_TO_REACH_PASSENGER_SERVICE.format());
    }
}

File: rating-service/src/main/kotlin/com/example/rating_service/client/PassengerServiceClientFallback.kt
package com.example.rating_service.client

import com.example.rating_service.util.ExceptionMessages
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Component

@Component
class PassengerServiceClientFallback: PassengerServiceClient {
    override fun doesPassengerExist(passengerId: Long): ResponseEntity<Boolean> {
        throw RuntimeException(ExceptionMessages.UNABLE_TO_REACH_PASSENGER_SERVICE)
    }
}